admin-dev/themes/new-theme/scss/components/_grid.scss

```diff
@@ -93,6 +93,9 @@ table .column-filters td:last-child .grid-reset-button {
             display: inline-block;
             width: 60px;
           }
+          .d-inline-block.md-checkbox ~ .d-inline-block.form-control {
+            width: calc(100% - 29px);
```

Comment:
Maybe you could use `width: fit-content;` ?

Reply:
https://caniuse.com/#search=fit-content

I think no as it represent almost 4% of the web traffic (and as @matks think, maybe 8% 🤣 ) where it will break